packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanTreeOffset.js

```diff
@@ -14,33 +14,145 @@
 import React from 'react';
 import IoChevronRight from 'react-icons/lib/io/chevron-right';
 import IoIosArrowDown from 'react-icons/lib/io/ios-arrow-down';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import { actions } from './duck';
+
+import type { ReduxState } from '../../../types/index';
+import type { Span } from '../../../types/trace';
 
 import './SpanTreeOffset.css';
 
-type SpanTreeOffsetProps = {
-  level: number,
+type SpanTreeOffsetOwnPropsType = {
+  addHoverIndentGuideId: string => void,
+  removeHoverIndentGuideId: string => void,
   hasChildren: boolean,
   childrenVisible: boolean,
+  span: Span,
   onClick: ?() => void,
 };
 
-export default function SpanTreeOffset(props: SpanTreeOffsetProps) {
-  const { level, hasChildren, childrenVisible, onClick } = props;
-  const wrapperProps = hasChildren ? { onClick, role: 'switch', 'aria-checked': childrenVisible } : null;
-  const icon = hasChildren && (childrenVisible ? <IoIosArrowDown /> : <IoChevronRight />);
-  return (
-    <span className={`SpanTreeOffset ${hasChildren ? 'is-parent' : ''}`} {...wrapperProps}>
-      <span style={{ paddingLeft: `${level * 20}px` }} />
-      {icon && <span className="SpanTreeOffset--iconWrapper">{icon}</span>}
-    </span>
-  );
+type SpanTreeOffsetPropsType = SpanTreeOffsetOwnPropsType & {
+  hoverIndentGuideIds: Set<string>,
+};
+
+export class UnconnectedSpanTreeOffset extends React.PureComponent<SpanTreeOffsetPropsType> {
+  ancestorIds: string[];
+  props: SpanTreeOffsetPropsType;
+
+  static defaultProps = {
+    childrenVisible: false,
+    hasChildren: false,
+    onClick: null,
+  };
+
+  constructor(props: SpanTreeOffsetPropsType) {
+    super(props);
+
+    this.ancestorIds = [];
+    let currentSpan: Span = props.span;
+    while (currentSpan) {
+      currentSpan = _get(_find(currentSpan.references, { refType: 'CHILD_OF' }), 'span');
+      if (currentSpan) {
+        this.ancestorIds.push(currentSpan.spanID);
+      }
+    }
+
+    // Some traces have multiple root-level spans, this connects them all under one guideline and adds the
+    // necessary padding for the collapse icon on root-level spans.
+    this.ancestorIds.push('root');
+
+    this.ancestorIds.reverse();
+  }
+
+  /**
+   * If the mouse leaves to anywhere except another span with the same ancestor id, this span's ancestor id is
+   * removed from the set of hoverIndentGuideIds.
+   *
+   * @param {Object} event - React Synthetic event tied to mouseleave. Includes the related target which is
+   *     the element the user is now hovering.
+   * @param {string} ancestorId - The span id that the user was hovering over.
+   */
+  handleMouseLeave = (event: SyntheticMouseEvent<HTMLSpanElement>, ancestorId: string) => {
+    if (
+      !(event.relatedTarget instanceof HTMLSpanElement) ||
+      // !event.relatedTarget.getAttribute ||
+      event.relatedTarget.getAttribute('data--ancestor-id') !== ancestorId
```

Comment:
If you change the attribute to `data-ancestor-id` you can use:

```js
event.relatedTarget.dataset.ancestorId
```

[MDN data attribute name conversion](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#Name_conversion)

Reply:
nice, that definitely feels like a more correct usage than assigning and reading an arbitrary attribute.